NiBAx/plugins/harmonization/harmonization.py

```diff
@@ -378,8 +378,19 @@ def DoHarmonization(self):
 
         if 'isTrainMUSEHarmonization' in self.datamodel.data.columns:
             muse = pd.concat([self.datamodel.data['isTrainMUSEHarmonization'].copy(), covars, pd.DataFrame(bayes_data, columns=['H_' + s for s in self.datamodel.harmonization_model['ROIs']])],axis=1)
+        if 'UseForComBatGAMHarmonization' in self.datamodel.data.columns:
+            muse = pd.concat([self.datamodel.data['UseForComBatGAMHarmonization'].copy(), covars, pd.DataFrame(bayes_data, columns=['H_' + s for s in self.datamodel.harmonization_model['ROIs']])],axis=1)
         else:
             muse = pd.concat([covars,pd.DataFrame(bayes_data, columns=['H_' + s for s in self.datamodel.harmonization_model['ROIs']])],axis=1)
+        
+        # harmonize derived volumes 
+        _, MUSEDict = self.datamodel.GetMUSEDictionaries()
+        muse_mappings = MUSEDict.reset_index().rename(columns={'level_0':'ROI_INDEX','level_1':'ROI_NAME'}).set_index('ROI_INDEX').drop(columns=['ROI_NAME'])
```

Comment:
@melhemr This line caused an error as `MUSEDict` is not a data frame, but a dictionary.

Reply:
Can I hardcode reference MUSE_ROI_Dictionary.csv in the data plugin? 

I'm not sure why you are getting an error, but I don't when I run the code.

Reply:
@melhemr `MUSEDict` is a `dict` object and as such has no attribute `reset_index`. It also has no information about whether the ROI was `DERIVED` or `SINGLE`. Please check the definition of `GetMUSEDictionaries()` to understand what the function returns. It is not the data frame that contains `ROI_INDEX`, `ROI_NAME`, and `ROI_LEVEL` but only two dictionaries to translate from index to name and vice-versa.

Please use the same approach to locate `MUSE_ROI_Dictionary.csv` as in `GetMUSEDictionaries()`. To solve the issue you could add a third output to `ReadMUSEDictionary()` which is the entire data frame (i.e. `MUSE_Dict` in the definition of `ReadMUSEDictionary()`) and extend `SetMUSEDictionaries()` to also write the data frame as field.